### ISSUE TITLE
Fix broken UnreadIndicatorsTest

### ISSUE THREAD

**Issue by roryabraham** [Expensify/App#14352]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

### Problem
[This PR](https://github.com/Expensify/App/pull/14224) broke the `UnreadIndicatorsTest`. It did not get flagged on the PR because [the test was not run for some reason](https://github.com/Expensify/App/actions/runs/3896726517) (treating this as a separate problem and planning to deal with it by completing https://github.com/Expensify/App/issues/14088 before creating another issue). 

### Solution
This test is failing because local notifications used to be triggered by Onyx updates, but https://github.com/Expensify/App/pull/14224 changes that so they are triggered by Pusher events, which trigger both a notification and an Onyx update.

However, `UnreadIndicatorsTest` is currently structured only to mock Onyx data updates, not Pusher events. We need to change that so it mocks Pusher events instead.

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by mountiny** [comment]
Merged a PR to skip over this test https://github.com/Expensify/App/pull/14363 but we still need a real fix to be implemented.

**Comment by arosiclair** [comment]
PR with proper fixes is out. Since we already pushed a hotfix that skips the test, I'll drop the priority on this a little

**Comment by arosiclair** [comment]
This was pushed to prod a couple days ago

**Comment by arosiclair** [comment]
Hey @twisterdotcom you mind handling payment for @thesahindia?

**Comment by twisterdotcom** [comment]
Yes, I haven't created a manual Upwork job before and I'm OOO tomorrow. @thesahindia do you mind if this comes later in the week? I can reassign if not.

**Comment by MelvinBot** [comment]
@arosiclair @twisterdotcom this issue is now 3 weeks old. There is one more week left before this issue <a href='https://stackoverflowteams.com/c/expensify/questions/15233/15234#15234'>breaks WAQ</a>. What needs to happen to get a PR in review this week? Please create a thread in #expensify-open-source to discuss. Thanks!

**Comment by twisterdotcom** [comment]
Shh Melv. I just need to pay out.

**Comment by twisterdotcom** [comment]
Contract extended. Just waiting on you to accept (this and the other one Sahil!)

**Comment by twisterdotcom** [comment]
Paid!
